### ISSUE TITLE
Action bar title is slightly offset from the material design guidelines

### ISSUE THREAD

**Issue by barbeau** [OneBusAway/onebusaway-android#837]
**Summary:** 

After updating to `compileSdkVersion 26`, the action bar title shifted slightly to the right.  Now, it's no longer exactly aligned with the lists - for example, starred stops:

![image](https://user-images.githubusercontent.com/928045/34314332-8b7f3ef0-e740-11e7-8901-c9c5bbdc3ae9.png)

Prior to `compileSdkVersion 26` the "Starred stops" title was exactly aligned with the stop names (i.e., at the red line).

This may be related to the fact that we're still using the deprecated `ActionBarCompat` instead of the toolbar.

**Steps to reproduce:** 

1. Star a stop
1. Go to the "Starred stops" view in the main HomeActivity

**Expected behavior:** 

Action bar title should be exactly aligned with the stop names

**Observed behavior:** 

Action bar title has shifted right slightly and is no longer aligned with the stop name

**Device and Android version:** 

Samsung Galaxy S8+ w/ Android 8.0 beta

**Screenshots:** 

See above

**Comment by fireblue1** [comment]
Just started working on this issue!

**Comment by barbeau** [comment]
Thanks @fireblue1!

**Comment by fireblue1** [comment]
**Update:** An instance of this issue can be found within the linear layout of `stop_list_item.xml` (./root/res/layout/stop_list_item.xml'). Within the ViewGroup, there exists an attribute called `android:layout_marginLeft="@dimen/keyline_2" `. Within `dimens.xml` the variable `keyline_2` given a value of `40dp`. This value fails to properly align the text in the saved stops tab to that in the action bar. I went ahead and experimented with a variety of sizes to match the alignment and found that 52dp works best (please see the image attached). I will go ahead and change the value of the `keyline_2` dimen and observe how it affects other lists within the application.
![52dp_dimen](https://user-images.githubusercontent.com/26015439/42486940-a19562cc-83cc-11e8-8c77-93298e64fe3e.png)

**Comment by barbeau** [comment]
@fireblue1 while you're working on this, could you please take a look at the material design layout specs, in particular:
* https://material.io/design/layout/responsive-layout-grid.html#columns-gutters-margins
* https://material.io/design/layout/spacing-methods.html#spacing

We should try to confirm to the standard spacing, which I believe the current keylines do. 

I've wondered if the Action bar title spacing is actually the one that's wrong, perhaps because we're using the deprecated Action Bar and not a Toolbar?

**Comment by fireblue1** [comment]
@barbeau I'm noting the specifications from the Material Design standards that you posted and also those for the App Bar:

- https://material.io/design/components/app-bars-top.html#specs

The current keylines are correct as following the Material guidelines, so modifying the position of text within the app bar may be the most appropriate solution. I know that the toolbar view group is a more extensive in terms of attribute customization than the Action Bar Compact. I'm attempting to locate the xml configuration for the current action bar within OneBusAway, but so far I've only been had luck finding the code for the settings and about pages. Documentation is available from the Android Development Blogs on the process to switch from using to old Action Bar to the Toolbar which can be found [here](https://blog.xamarin.com/android-tips-hello-toolbar-goodbye-action-bar/).

**Comment by barbeau** [comment]
Ok, thanks, that's good to know. 

I'm on mobile right now, but IIRC all ActionBar setup is done through:
https://github.com/OneBusAway/onebusaway-android/blob/master/onebusaway-android/src/main/java/org/onebusaway/android/util/UIUtils.java#L123

Also, a while back I did a quick implementation to switch to Toolbar, but it had some strange effects with other things in main.xml layout like the route header. So it will likely take some tweaking to get everything working correctly together.

**Comment by fireblue1** [comment]
**Update #2: ** For the past few days, I messed around with trying to switch to the Toolbar from Action Bar Compat, but I ran into a lot of issues. Then I realized that the `ActionBar` object has a method called `setCustomView` that quite literally permits a custom view to be "merged" with the current Action Bar. However, none of this is really necessary because the issue here comes down to some mathematics. When @barbeau reminded me about the Material Design standards, there was something that, again, appeared strange about the value for `keyline_2` within `dimens.xml`. I revisited the specification images that the Material Design guideline has for [Keylines](https://material.io/design/layout/spacing-methods.html#spacing) and [Action Bars](https://material.io/design/components/app-bars-top.html#specs). Notice that the text in the application's activity and action bar are aligned at 72dp. The text in the action bar in OneBusAway is also aligned in 72dp, yet not in lists, like Starred stops and Reminders. The reason for this inconsistency comes from an understanding of how the left margin is formed. We have a formula essentially, **`left_margin = keyline_1 + keyline_2`**. With the current dimens for `keyline_1` of 16dp and `keyline_2 `of 40dp we end up with a left margin of 56dp. An adjustment of `keyline_2` to 56dp gives us our desired total of 72dp for the left margin.

**Comment by barbeau** [comment]
Thanks for the update! Yes, if we have an error in the keylines when combined with other layout dimensions please fix. Obviously, we'll need to look at everywhere the dimen is used to make sure the change doesn't negatively affect other layouts. 

Also it's a little late now but I just realized that in another app I maintain I am indeed using Toolbar:
* https://github.com/barbeau/gpstest/blob/master/GPSTest/src/main/res/layout/activity_main.xml
* https://github.com/barbeau/gpstest/blob/master/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java

The main layout and navigation drawer are a very similar implementation to OBA and share some code.

We should eventually switch to Toolbar even if we can solve these issues with correct keylines dimen, but we can always push that to another issue.

**Comment by fireblue1** [comment]
I will continue to test the application with the new keyline dimensions and will search the code for every view group that it is utilized in. I really do agree with you that the acton bar needs to updated to the Toolbar for even just the sake of future compatibility and customization within Android Studio. I think creating a basic Toolbar view group isn't too troubling. However, issues arose when I attempted to modify the setupActionBar method in `UIUtils.java`. The application would build successfully, but crash upon startup every time. I know that this method is used in several of the UI activities and I probably need to considering passing the current activity to the Toolbar object in a similar way the setupActionBar method does so currently. Thank you for the code from the Toolbar that you are using in another app, as there's not really a lot out there as to how to change an app which makes extensive use of an action bar to a Toolbar. I will work on this and let you know my progress. (I can continue to post updates here if you would like or in another issue dedicated to this if that is better).

**Comment by GajananPalepwad** [comment]
@barbeau please do check my pr I have solved this issue

**Comment by barbeau** [comment]
@aaronbrethorst is leading development on OBA Android now, so CCing him.

**Comment by GajananPalepwad** [comment]
@aaronbrethorst please do check my pr I have solved this issue

**Comment by aaronbrethorst** [comment]
I don't think we should fix this anymore. Android UI style has moved away from the current design and I think any solution we come up with will have some serious tradeoffs.
